mapeditor/playersettings.cpp

```diff
@@ -0,0 +1,77 @@
+#include "StdInc.h"
+#include "playersettings.h"
+#include "ui_playersettings.h"
+#include "playerparams.h"
+#include "mainwindow.h"
+
+PlayerSettings::PlayerSettings(MapController & ctrl, QWidget *parent) :
+	QDialog(parent),
+	ui(new Ui::PlayerSettings),
+	controller(ctrl)
+{
+	ui->setupUi(this);
+	show();
+
+	int players = 0;
+	for(auto & p : controller.map()->players)
+	{
+		if(p.canAnyonePlay())
+		{
+			paramWidgets.push_back(new PlayerParams(controller, players));
+			ui->playersLayout->addWidget(paramWidgets.back());
+			++players;
+		}
+	}
+
+	if(players < 2)
+		ui->playersCount->setCurrentText("");
+	else
+		ui->playersCount->setCurrentIndex(players - 2);
+	
+	setAttribute(Qt::WA_DeleteOnClose);
+}
+
+PlayerSettings::~PlayerSettings()
+{
+	delete ui;
+}
+
+void PlayerSettings::on_playersCount_currentIndexChanged(int index)
+{
+	assert(index + 2 <= controller.map()->players.size());
```

Comment:
will it work?

auto selectedPlayerCount = index + 2;

and use it everywhere?